src/env/environment.rs

```diff
@@ -452,6 +454,41 @@ fn get_hostname_identifier() -> Option<WString> {
     }
 }
 
+/// Get values for $USER and $HOME via getpwuid,
+/// without trusting $USER or $HOME.
+pub fn get_user_home() -> (Option<String>, Option<String>) {
```

Comment:
The "user" return value is unused, so it's better to remove it.

Again, we can avoid requiring UTF-8 here (unlike much of the Rust ecosystem)
by returning a `Vec<u8>`, which is suitable for consumption by `str2wcstring`